patcher/Patcher.py

```
@@ -324,51 +292,37 @@ def align(v, a):
+    create_sections_file(build_folder_path / "section.ld",
+                         function_addresses | config.functions)
     if run_system(
-            f"""cd {build_folder_path} &
-            {gcc_compiler_path} {GCC_FLAGS}
-            -I ../include/ {includes}
-            -Wl,-T,../section.ld,--image-base,{base_pe.imgbase + new_v_offset - 0x1000},-s,-Map,sectmap.txt,-o,section.pe
-            ../section/main.cpp"""):
-        raise Exception("Errors occurred during building of patch files")
+        t""" cd {build_folder_path} &
+            {config.gcc_path} {" ".join(config.gcc_flags)}             -I {include_folder_path}             -Wl,-T,section.ld,--image-base,{image_base},-s,-Map,sectmap.txt,-o,section.pe             {section_folder_path / "main.cpp"}"""):
+                raise Exception("Errors occurred during building of patch files")
```

Ruff (0.14.14)

[warning] Create your own exception

(TRY002)